sklearn/preprocessing/data.py

```diff
@@ -788,9 +788,7 @@ def transform(self, X, y=None):
             warnings.warn(DEPRECATION_MSG_1D, DeprecationWarning)
 
         if sparse.issparse(X):
-            if X.shape[0] == 1:
-                inplace_row_scale(X, 1.0 / self.scale_)
-            else:
+            if self.scale_ is not None:
```

Comment:
you can remove this check. If you scroll up you can see that the check has already been done :)